app.py

```diff
@@ -33,6 +36,64 @@
 # Set the secret key to enable access to session data.
 app.secret_key = os.urandom(24)
 
+client = datastore.Client()
+
+GENIUS_ACCESS_TOKEN = os.getenv('GENIUS_ACCESS_TOKEN')
+
+@app.route('/home')
+def recommend():
+    """Recommend a song"""
+    headers = {
+        'Authorization': 'Bearer {token}'.format(token=GENIUS_ACCESS_TOKEN)}
+
+    # Hard-coding a song recommendation
+    url = 'https://api.genius.com/'
+    endpoint = 'songs/'
+
+    # Song ID for 'Space Song' by Beach House
+    song_id = '1929412'
+
+    response = requests.get(
+        url + endpoint + song_id,
+        headers=headers)
+
+    song = response.json()['response']['song']
+
+    return flask.render_template('index.html', song=song)
+
+@app.route('/likes')
+def get_likes():
+    """Get liked songs"""
+    kind = 'LikedSong'
+
+    query = client.query(kind=kind)
+    songs = list(query.fetch())
+
+    return flask.render_template('likes.html', songs=songs)
+
+@app.route('/like', methods=['POST'])
+def add_like():
+    """Like a song"""
+    if flask.request.method == 'POST':
```

Comment:
because you've specified methods=POST, flask.request.method will be POST (so you can remove this)

otherwise, try to use guard clauses to eliminate nesting:

```py
if flask.request.method != 'POST':
  return flask.abort(400)

..rest of code de-dented
```